src/macaron/slsa_analyzer/git_service/gitlab.py

```diff
@@ -117,7 +123,74 @@ def clone_repo(self, clone_dir: str, url: str) -> None:
             If there is an error cloning the repository.
         """
         clone_url = self.construct_clone_url(url)
-        git_url.clone_remote_repo(clone_dir, clone_url)
+        repo = git_url.clone_remote_repo(clone_dir, clone_url)
```

Comment:
If a `CloneError` exception is raised by  `git_url.clone_remote_repo`, we might miss resetting the remote, right?

Reply:
From the `git_url.clone_remote_repo` function, it's mentioned that a `CloneError` exception is raised whenever the repository has not been cloned or the clone attempts failed. In both cases, the repository would not be available so we could update the origin remote URL.

Reply:
Might be useful to add a comment to clarify.

Reply:
Fixed in https://github.com/oracle/macaron/pull/354/commits/6678fe22d14923dfb62282973f68e093b5abd38b